zproject/backends.py

```diff
@@ -985,7 +985,7 @@ def __init__(self, *, user_profile: Optional[UserProfile]=None,
                                                                 "with login_token is disallowed.")
             self.instantiate_with_token(login_token, delete_stored_data)
         else:
-            self.data_dict = cast(ExternalAuthDataDict, {**data_dict})
+            self.data_dict = data_dict.copy()
```

Comment:
Should this be `ExternalAuthDataDict(**data_dict)`?  Seems like we want the type for this object to be that.

Reply:
It is that; Mypy knows `data_dict.copy()` is an `ExternalAuthDataDict`, while `ExternalAuthDataDict(**data_dict)` is an error.